_posts/2017-06-19-How-to-create-a-shared-sstate-dir.md

```diff
@@ -0,0 +1,91 @@
+---
+layout: post
+title:  "How to create a shared sstate and downloads directory?"
+date:   2017-06-19 14:48:00 +0200
+---
+
+By reusing the `sstate-cache` and `downloads` directories the image build
+turnaround time will be decreased. From the Yocto Project reference
+manual:
+
+> The Yocto OpenEmbedded build system builds everything from scratch unless
+> BitBake can determine that parts do not need to be rebuilt. The Yocto
+> Project implements shared sstate code that supports incremental builds.
+> The implementation of the shared state code answers the following
+> questions that were fundamental roadblocks within the OpenEmbedded
+> incremental build support system:
+>   * What pieces of the system have changed and what pieces have not changed?
+>   * How are changed pieces of software removed and replaced?
+>   * How are pre-built components that do not need to be rebuilt from scratch used when they are available?
+
+## Setting up a NFS server
+
+Half of the problem is setting up the infrastructure to support the shared
+sstate. This section describes how to set up a shared sstate directory on
+a server using NFS.
+
+To install the NFS dependencies run the following command:
+
+```
+sudo apt-get update && sudo apt-get install nfs-kernel-server nfs-common
+```
+
+By editing the `/etc/exports` file we instruct the NFS server what
+directory is being shared and on which subnet:
+
+```
+sudo echo "/path/to/sstate-cache 192.168.1.1/24(rw,sync,no_root_squash,no_subtree_check)" >> /etc/exports"
+```
+
+Setting up the `sstate-cache` directory as read/write or
+readonly is dependent on the use case. If there is a primary host that
+populates the `sstate-cache`, while the other nodes should reuse and not
+populate the directory then it should be mounted readonly.
+
+## Setting up a NFS client
+
+Mount a shared NFS directory by executing the following command:
+
+```
+sudo mount example.server:/path/to/sstate-cache /local/mounted/shared-sstate-cache
+```
+
+or alternatively populate the `/etc/fstab` as shown below. This will
```

Comment:
Wasn't this because of enabling slaves to write to the cache during incremental builds? Which is a design I think we said we would not have at this point. I think we agreed that a nightly clean build would create the cache and that the incremental builds would just use that "read only". The downside would be big tasks like rebuilding Qt where everyone would need to rebuild all of it during the day between the introduction of the change and the next nightly. Something we can mitigate by e.g. manually triggering a clean "nightly" that would update the cache.

IMHO it would be better with as clean and simple setup as possible at this point and add optimizations later if needed.